### PR TITLE
Prevent publishing metrics to AppOptics when having no metrics

### PR DESCRIPTION
**Pull request by izeye**
This PR changes to prevent publishing metrics to AppOptics when having no metrics.

Closes gh-1235

### CODE REVIEW

File: implementations/micrometer-registry-appoptics/src/main/java/io/micrometer/appoptics/AppOpticsMeterRegistry.java [hunk 1]
```diff
@@ -96,24 +96,27 @@ public void start(ThreadFactory threadFactory) {
...
+                                this::writeLongTaskTimer,
+                                this::writeTimeGauge,
+                                this::writeFunctionCounter,
+                                this::writeFunctionTimer,
+                                this::writeMeter)
+                        )
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(joining(",", "{\"measurements\":[", "]}"));
+                if (body.equals("{\"measurements\":[]}")) {
```
Comment (CandiedCode) [inline implementations/micrometer-registry-appoptics/src/main/java/io/micrometer/appoptics/AppOpticsMeterRegistry.java]: @izeye  Thanks for jumping on this so quick!

does it make sense to have a Static String for this?  

```
private Static final String EMPTY_MEASURES_BODY = "{\"measurements\":[]}"; 
...
if (body.equals(EMPTY_MEASURES_BODY)) {

Comment (izeye) [reply implementations/micrometer-registry-appoptics/src/main/java/io/micrometer/appoptics/AppOpticsMeterRegistry.java]: @CandiedCode Thanks for the feedback! I updated in d9377b8 as suggested.

Comment (shakuzen) [reply implementations/micrometer-registry-appoptics/src/main/java/io/micrometer/appoptics/AppOpticsMeterRegistry.java]: From the bytecode perspective, it is a constant string either way. If we want to try to make sure we don't update one part and not the other (resulting in a bug), it'd be better to make the prefix and suffix constants and then use the concatenation of those in the `equals` method.

Comment (izeye) [reply implementations/micrometer-registry-appoptics/src/main/java/io/micrometer/appoptics/AppOpticsMeterRegistry.java]: @shakuzen Sounds good to me. I pushed d90696f.

**Pull request merged**